A bunch of assorted fixes of covscan reported issues

alexey-tikhonov opened the pull request:
Issues were found using custom model by @stevegrubb

aplopez commented on src/providers/krb5/krb5_ccache.c:
```diff
@@ -637,12 +637,12 @@ errno_t get_ccache_file_data(const char *ccache_file, const char *client_name,
     krb5_free_cred_contents(ctx, &cred);
 
     kerr = krb5_cc_close(ctx, cc);
+    cc = NULL;
```
Well, maybe there is nothing we can do with `cc` once `krb5_cc_close()` failed. Is that the case?

alexey-tikhonov replied:
`krb5_cc_close()` is a cleanup function. If it fails, there is nothing else can be done.
If 'cc' isn't set to NULL here then it case of error there is a 2nd attempt of `krb5_cc_close()` in 'done:' section that doesn't make sense at best (or is kind of 'double free' at worst).

aplopez replied:
OK

alexey-tikhonov commented on src/providers/krb5/krb5_child.c:
```diff
@@ -1400,6 +1400,11 @@ static krb5_error_code create_ccache(char *ccname, krb5_creds *creds)
         krb5_cc_close(kctx, kcc);
     }
+
+    if (kctx != NULL) {
```
Removed `if (kctx != NULL)` condition.

aplopez reviewed (approved):
ACK. Thanks for the patch.

thalman reviewed (approved):
LGTM, thank you for the patch

alexey-tikhonov commented:
Pushed PR: https://github.com/SSSD/sssd/pull/6792

* `master`
    * d02533caca667b51f29fa02ee9ed48c8b3896c69 - UTILS: fixed USE_AFTER_FREE
    * eca00ef4719c44c4e68ead3346a16229b6471d13 - LDAP: fixed leak of `kprinc`
    * fd7da517ddd0e220f081ad9e7b5d7fcb0cae39b7 - LDAP: fixed RESOURCE_LEAK
    * 01f0d067f1e4ba8ec3710f515d21631a53c9c9ef - KRB5: fixed RESOURCE_LEAK
    * a83be8fb51172d4e1a282a0a078d81ee93afdcb5 - KRB5: avoid RESOURCE_LEAK
    * 75822701770179582c344960603cce8bd54a7890 - KRB5: fix memory leak
    * b69ff375a2b185219bae91c48aa7bfb3138b98f2 - KRB5: fix memory leak
    * 7f308c6fe01408fa6beb48b9f7627068968da771 - KRB5: avoid FORWARD_NULL
    * ff5096bb766765e45aaad156285a603a21aa1bc8 - KRB5: use proper function to deallocate mem
    * f6bbd591d636e4309ec37659f825b0f9c53d4b6b - KRB5: avoid another attempt to free 'cc' in 'done:' section if first attempt failed.
